packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/ExpoBridgeModule.kt

```diff
@@ -8,15 +9,33 @@ import java.lang.ref.WeakReference
 /**
  * The classic bridge module that is responsible for installing the host object to the runtime.
  */
-class ExpoBridgeModule(private val nativeModulesProxy: WeakReference<NativeModulesProxy>) : ReactContextBaseJavaModule() {
+class ExpoBridgeModule(
+  reactContext: ReactApplicationContext,
+  private val nativeModulesProxy: WeakReference<NativeModulesProxy>
+) :
+  ReactContextBaseJavaModule(reactContext) {
   override fun getName(): String = "ExpoModulesCore"
 
   @ReactMethod(isBlockingSynchronousMethod = true)
   fun installModules(): Boolean {
+    // Bridgeless ReactHostImpl may have BridgelessReactContext ready but not ReactInstance.
+    // Try to busy wait until ReactInstance is available so we could get the javaScriptContextHolder.
+    tryWaitSync(waitMs = 100, retries = 10) {
+      reactApplicationContext.hasActiveReactInstance()
```

Comment:
Why can we run a js method when the application context isn't ready? It's looking like a bug in the React native core. Regardless, let's merge it for now, although I don't like solutions like this 😅

Reply:
i think that's a bug inside ReactHostImpl where all [mReactInstanceTaskRef.get().getResult()](https://github.com/facebook/react-native/blob/99d0c540cac42efec2ab5943b8d0510bbf94e16b/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/ReactHostImpl.java#L551-L573) return null. i would raise the bug to meta.
would merge this first and remove it once meta resolve the issue.